backup/statistics.go

```diff
@@ -69,14 +69,16 @@ func GenerateAttributeStatisticsQueries(table Table, attStat AttributeStatistic)
 		attributeSlotsQueryStr = generateAttributeSlotsQuery4(attStat)
 	}
 
-	attributeQueries = append(attributeQueries, fmt.Sprintf(`DELETE FROM pg_statistic WHERE starelid = %s AND staattnum = %d;`, starelidStr, attStat.AttNumber))
-	attributeQueries = append(attributeQueries, fmt.Sprintf(`INSERT INTO pg_statistic VALUES (
+	attributeQueries = append(attributeQueries, fmt.Sprintf(`DELETE FROM pg_statistic
+WHERE starelid = %s AND staattnum = (SELECT attnum FROM pg_attribute WHERE attrelid = %s AND attname = '%s');`, starelidStr, starelidStr, utils.EscapeSingleQuotes(attStat.AttName)))
+	attributeQueries = append(attributeQueries, fmt.Sprintf(`INSERT INTO pg_statistic SELECT
 	%s,
-	%d::smallint,%s
+	attnum,%s
 	%f::real,
 	%d::integer,
 	%f::real,
-	%s);`, starelidStr, attStat.AttNumber, inheritStr, attStat.NullFraction, attStat.Width, attStat.Distinct, attributeSlotsQueryStr))
+	%s
+FROM pg_attribute WHERE attrelid = %s AND attname = '%s';`, starelidStr, inheritStr, attStat.NullFraction, attStat.Width, attStat.Distinct, attributeSlotsQueryStr, starelidStr, utils.EscapeSingleQuotes(attStat.AttName)))
```

Comment:
Can we guarantee that exactly one row will be inserted here?

Reply:
there can't be duplicate values for a given attrelid and attname.

Reply:
What I meant was that we won't insert zero rows.

Reply:
> What I meant was that we won't insert zero rows.

Zero rows could be if there is no such column or table.
But metadata is backed up before the statistics, so it looks that for every statistics value there should be a corresponding pair of table&column in the metadata.
Thus the same tables with same columns should appear it the restored database.

Reply:
> metadata is backed up before the statistics

what if
```sh
gprestore --data-only --with-stats
```

Reply:
In this case if there is a non-existing column, than we will insert 0 rows into the pg_statistic (which I think is what expected - no junk statistic for non-existing columns).

If there is a missing table, than expression `'<schema>.<relation>'::regclass::oid` will error out, but that is the same behavior as it was before my patch (as it was also used in old queries). So it is a separate issue.

Reply:
Before the patch, this situation was possible:
```sql
insert into t values (1, 2, 3)
```
and after the patch it became:
```sql
insert into t select 1, 2, 3 where false
```
How justified is this?

Reply:
we have "where false" in case there is no suitable <table> & <column> pair. We shouldn't insert any data in this case.